kornia/geometry/liegroup/se2.py

```diff
@@ -0,0 +1,235 @@
+        """Constructor for the base class.
+
+        Internally represented by a complex number `z` and a translation 2-vector.
+
+        Args:
+            r: So2 group encompassing a rotation.
+            t: translation vector with the shape of :math:`(B, 2)`.
+
+        Example:
+            >>> so2 = So2.identity(1)
+            >>> t = torch.ones((1, 2))
+            >>> se2 = Se2(so2, t)
+            >>> se2
+            rotation: Parameter containing:
+            tensor([1.+0.j], requires_grad=True)
+            translation: Parameter containing:
+            tensor([[1., 1.]], requires_grad=True)
+        """
+        super().__init__()
+        KORNIA_CHECK_TYPE(r, So2)
+        # TODO change to KORNIA_CHECK_SHAPE once there is multiple shape support
+        check_se2_r_t_shape(r, t)
+        self._r = r
+        self._t = Parameter(t)
+
+    def __repr__(self) -> str:
+        return f"rotation: {self.r}\ntranslation: {self.t}"
+
+    def __getitem__(self, idx) -> 'Se2':
+        return Se2(self._r[idx], self._t[idx][None])
+
+    def __mul__(self, right: "Se2") -> "Se2":
+        """Compose two Se2 transformations.
+
+        Args:
+            right: the other Se2 transformation.
+
+        Return:
+            The resulting Se2 transformation.
+        """
+        if isinstance(right, Se2):
+            KORNIA_CHECK_TYPE(right, Se2)
+            so2 = self.so2
+            r = so2 * right.so2
+            t = self.t + so2 * right.t
+            return Se2(r, t)
+        elif isinstance(right, Tensor):
+            KORNIA_CHECK_TYPE(right, Tensor)
+            # TODO change to KORNIA_CHECK_SHAPE once there is multiple shape support
+            check_se2_r_t_shape(self.so2, right)
+            return self.so2 * right + self.t
+        else:
+            raise TypeError(f"Unsupported type: {type(right)}")
+
+    @property
+    def so2(self) -> So2:
+        """Return the underlying rotation(So2)."""
+        return self._r
+
+    @property
+    def r(self) -> So2:
+        """Return the underlying rotation(So2)."""
+        return self._r
+
+    @property
+    def t(self) -> Tensor:
+        """Return the underlying translation vector of shape :math:`(B,3)`."""
+        return self._t
+
+    @staticmethod
+    def exp(v) -> 'Se2':
+        """Converts elements of lie algebra to elements of lie group.
+
+        Args:
+            v: vector of shape :math:`(B, 3)`.
+
+        Example:
+            >>> v = torch.ones((1, 3))
+            >>> s = Se2.exp(v)
+            >>> s.r
+            Parameter containing:
+            tensor([0.5403+0.8415j], requires_grad=True)
+            >>> s.t
+            Parameter containing:
+            tensor([[0.3818, 1.3012]], requires_grad=True)
+        """
+        check_v_shape(v)
+        theta = v[..., 2]
+        so2 = So2.exp(theta)
+        z = tensor(0.0, device=v.device, dtype=v.dtype)
+        a = where(theta != 0.0, so2.z.imag / theta, z)
```

Comment:
create a single variable `theta != 0.0` -- we don't want the same operation to happen twice